### PR TITLE
Finish string conversion and add list parsing

### PR DESCRIPTION
**Pull request by jcdyer**
Clean up and list parsing.
## Reviewers
- [x] @adampalay

### CODE REVIEW

File: jsonParser.hs [hunk 1]
```diff
@@ -3,15 +3,15 @@
 data JSON = N Double | S String | B Bool | Null | L [JSON] | O [(String, JSON)]
     deriving (Show, Eq)
 
-data Parser a = (a, String)
+type Parser a = (a, String)
```
Comment (adampalay) [inline jsonParser.hs:6]: why type vs data?

Comment (jcdyer) [reply jsonParser.hs:6]: I'm not sure why, but ghci was complaining about overriding a builtin operator `,`.  Or something like that.  Try changing it back and run it.  You'll get the error.

Comment (jcdyer) [reply jsonParser.hs:6]: P.S. I love the readme.

**Comment by adampalay** [comment]
👍

**Comment by jcdyer** [comment]
One thumb is probably enough for this repo...

**Pull request merged**